src/transport/SessionManager.cpp

```diff
@@ -459,16 +459,18 @@ void SessionManager::SecureGroupMessageDispatch(const PacketHeader & packetHeade
     CHIP_ERROR err = CHIP_NO_ERROR;
     PayloadHeader payloadHeader;
     SessionManagerDelegate::DuplicateMessage isDuplicate = SessionManagerDelegate::DuplicateMessage::No;
+    FabricIndex fabricIndex = 0; // TODO : remove initialization once GroupDataProvider->Decrypt is implemented
     // Credentials::GroupDataProvider * groups = Credentials::GetGroupDataProvider();
 
     VerifyOrExit(!msg.IsNull(), ChipLogError(Inet, "Secure transport received NULL packet, discarding"));
+    VerifyOrExit(!payloadHeader.NeedsAck(), ChipLogError(Inet, "Group message received with ack request, discarding"));
 
     // MCSP check
     if (packetHeader.IsSecureSessionControlMsg())
     {
         if (packetHeader.GetDestinationNodeId().HasValue() && packetHeader.HasPrivacyFlag())
         {
-            // TODO
+            // TODO : Implement MCSP check. Issue #11076
```

Comment:
This block starts with a VerifyOrExit that will throw away MCSP messages because all messages that request acks are discarded, and then ends up here, apparently to deal with these MCSP messages that we already threw away.

Things we expect of groupcast messages:

* will probably arrive via multicast IP (although we don't really care that much)
* will enclose session type == 1 to indicate group session
* will enclose the group key hash in session ID field
* will be encrypted with a group key
* will enclose source node ID
* will enclose group ID as destination ID
* will have C flag unset -> not a control message
* will have R flag unset -> no ACK

Things we expect of MCSP messages:

* will arrive via unicast IP
* will enclose session type == 1 to indicate group session
* will enclose the group key hash in session ID field
* will be encrypted with a group key
* will enclose source node ID
* will enclose destination node ID
* will have C flag set -> IS a control message
* will probably have R flag set -> yes, need to ACK

I would suggest leaving packetHeader.IsSecureSessionControlMsg() as the top-level check, as it was prior to this PR.  Then you can cleanly separate all of the groupcast validity checks from the MCSP checks.

In this case, VerifyOrExit(!payloadHeader.NeedsAck()) only makes sense for groupcast (packetHeader.IsSecureSessionControlMsg() == false).

Reply:
Removed the check and created #11425 to handle all of those check.